### PR TITLE
Trainer review

### PR DESCRIPTION
**Pull request by CosminDarius1**
Story 1 finished and waiting for comments from the trainer's review

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+require_once('src/Movie.php');
+$db = new PDO('mysql:host=db;dbname=collectorapp','root','password');
+$db->setAttribute(PDO::ATTR_ERRMODE,PDO::ERRMODE_EXCEPTION);
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE,PDO::FETCH_ASSOC);
+$query = $db->prepare("SELECT `name`,`year`,`director`,`genre` FROM `movies`");
+$query->execute();
+$allMoviesAsArrays = $query->fetchAll();
+$allMoviesAsObjects = [];
+foreach($allMoviesAsArrays as $movieAsArray) 
+{
```
Comment (norayna) [inline index.php]: the opening bracket of the foreach should be on the same line as the foreach

File: index.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    $movieAsObject = new Movie($movieAsArray['name'], $movieAsArray['director'], $movieAsArray['genre'], $movieAsArray['year']);
+    array_push($allMoviesAsObjects, $movieAsObject);
+};
```
Comment (norayna) [inline index.php]: no need for semicolon after }

File: index.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    <link rel="stylesheet" type="text/css" href="style.css">
+</head>
+<body>
+    <header> 
+        <h1>Favorite Movies</h1>
+    </header>
+    <section class="card-section">   
+        <?php foreach($allMoviesAsObjects as $movie) {
+            echo $movie->createCard();
+        }
```
Comment (norayna) [inline index.php]: better to put the foreach on a new line (and adjust indentations as a consequence)

File: src/Movie.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            return
+            "<div class='card'>" .
+            "<h2>" .  $this->name . "</h2>" . 
+            "<p>Director: " . $this->director . "</p>" .
+            "<p>Genre: " . $this->genre . "</p>" . 
+            "<p>Year: " . $this->year . "</p>" . 
+            "</div>";
+        }
+    }
+
```
Comment (norayna) [inline src/Movie.php]: from line 18 to 27, they are all indented 1 tab too deep

File: src/Movie.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        $this->name = $name;
+        $this->director = $director;
+        $this->genre = $genre;
+        $this->year = $year;
+    }
+
+    public function createCard(): string
+        {
+            return
+            "<div class='card'>" .
```
Comment (norayna) [inline src/Movie.php]: needs to be double quotes around card - in HTML, attributes should always use double quotes

File: style.css [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  letter-spacing: 2px;
+}
+html {
+  font-size: 15px;
+}
+body {
+  background-color: #000000;
+}
+header {
+  width: 100%;
```
Comment (norayna) [inline style.css]: this line shouldn't do anything i think

File: style.css [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  justify-content: center;
+}
+h1 {
+  color: #c0c0c0;
+}
+body {
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+}
```
Comment (norayna) [inline style.css]: get rid

File: style.css [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  padding: 0 20px;
+  width: 40%;
+  flex-direction: column;
+  display: flex;
+  background-color: #ff8c00;
+}
+@media screen and (max-width: 800px) {
+  .card-section {
+    flex-direction: column;
+  }
```
Comment (norayna) [inline style.css:40]: needs auto margins on sides to be centred after you remove display flex from body

File: tests/MovieTest.php [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+use PHPUnit\Framework\TestCase;
+
+class MovieTest extends TestCase {
+
+    public function testCreateCard_succesfull()
+    {
+        $movieTest = new Movie('Barbie', 'Greta Gerwig', 'Comedy', 2023);
+        $expectedOutput = 
+        "<div class='card'>" .
```
Comment (norayna) [inline tests/MovieTest.php]: again need double quotes around card, also make sure test still passes

**Review by norayna — CHANGES_REQUESTED** [review]
good job very clean!!

**Review by norayna — CHANGES_REQUESTED** [review]
almost there, great work!!

unit test fails though 😬

**Comment by CosminDarius1** [comment]
> almost there, great work!! 
Thank you ! :D

File: src/Movie.php [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        return
+        '<div class="card">' .
+        '<h2>' .  $this->name . '</h2>' . 
+        '<p>Director: ' . $this->director . '</p>' .
+        '<p>Genre: ' . $this->genre . '</p>' . 
+        '<p>Year: ' . $this->year . '</p>' . 
+        '</div>';
+    }
+    }
```
Comment (norayna) [inline src/Movie.php]: this last closing bracket needs to be un-indented completely

**Pull request merged**